lib/devise/models/confirmable.rb

```diff
@@ -169,6 +169,7 @@ def skip_reconfirmation!
         # in models to map to a nice sign up e-mail.
         def send_on_create_confirmation_instructions
           send_confirmation_instructions
+          skip_reconfirmation!
```

Comment:
@printercu You're right.

When we are creating an account, the e-mail doesn't exist. Then when we create the account with the "confirmable" feature, the `email` attribute is set.

When we are updating an account with an e-mail, but the previous e-mail is `nil`, it means the e-mail does not exists. For me, this situation is very similar when we are creating an account. Then it should have the same behavior, set `e-mail` not the `unconfirmed_email`.

Then it's a bug that needs to be fixed. Don't you wanna to open a PR with the test case and the patch?

Thank you for reporting! ✨